packages/elements/src/components/app/app.component.ts

```diff
@@ -177,9 +191,43 @@ export class MutationTestReportAppComponent extends LitElement {
   public static styles = [globals, unsafeCSS(theme), tailwind];
 
   public readonly subscriptions: Subscription[] = [];
+
   public connectedCallback() {
     super.connectedCallback();
     this.subscriptions.push(locationChange$.subscribe((path) => (this.path = path)));
+    this.initializeSSE();
+  }
+
+  private source: EventSource | undefined;
+
+  private initializeSSE() {
+    if (!this.sse) {
+      return;
+    }
+
+    this.source = new EventSource(this.sse);
+    this.source.addEventListener('mutant-tested', (event) => {
+      const newMutantData = JSON.parse(event.data as string) as Partial<MutantResult> & Pick<MutantResult, 'id' | 'status'>;
+      if (!this.report) {
+        return;
+      }
+
+      const theMutant = this.mutants.get(newMutantData.id);
+      if (theMutant === undefined) {
+        return;
+      }
+
+      for (const [prop, val] of Object.entries(newMutantData)) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        (theMutant as any)[prop] = val;
+      }
+
+      this.updateModel(this.report);
```

Comment:
How does this work? We've updated `theMutant`, how does it magically update the screen when we re-initialize `this.mutants` in `this.updateModel`?

Reply:
It rebuilds the `this.rootModel` with the changed `this.report`, which includes all the new information. However this method does not trigger the re-render. The `this.updateContext` method triggers the repaint. I'm not sure if this triggers a full repaint or a partial one (I should investigate that...)